pkg/oc/cli/cmd/rollout/retry.go

```diff
@@ -180,23 +207,38 @@ func (o RetryOptions) Run() error {
 			continue
 		}
 
-		patches := set.CalculatePatches([]*resource.Info{{Object: rc, Mapping: mapping}}, o.Encoder, func(info *resource.Info) (bool, error) {
+		patches := set.CalculatePatchesExternal([]*resource.Info{{Object: rc, Mapping: mapping}}, func(info *resource.Info) (bool, error) {
 			rc.Annotations[appsapi.DeploymentStatusAnnotation] = string(appsapi.DeploymentStatusNew)
 			delete(rc.Annotations, appsapi.DeploymentStatusReasonAnnotation)
 			delete(rc.Annotations, appsapi.DeploymentCancelledAnnotation)
 			return true, nil
 		})
 
 		if len(patches) == 0 {
-			kcmdutil.PrintSuccess(false, o.Out, info.Object, false, "already retried")
+			printer, err := o.Printer("already retried")
```

Comment:
I'm facing similar issue when reworking `cancel-build` and I strongly dislike this construct. We need to fix it afterwards.